File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/ledger/validations/IndividualModeOwnershipValidator.java
package com.decrux.pocketr.api.services.ledger.validations;

import com.decrux.pocketr.api.entities.db.ledger.Account;
import com.decrux.pocketr.api.exceptions.ForbiddenException;
import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class IndividualModeOwnershipValidator {

    public void validate(List<Account> nonOwnedAccounts, boolean isHouseholdMode) {
        if (!nonOwnedAccounts.isEmpty() && !isHouseholdMode) {
            throw new ForbiddenException("Cannot post to accounts not owned by current user in individual mode");
        }
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/ledger/validations/IndividualModeOwnershipValidator.kt
package com.decrux.pocketr.api.services.ledger.validations

import com.decrux.pocketr.api.entities.db.ledger.Account
import com.decrux.pocketr.api.exceptions.ForbiddenException
import org.springframework.stereotype.Component

@Component
class IndividualModeOwnershipValidator {
    fun validate(
        nonOwnedAccounts: List<Account>,
        isHouseholdMode: Boolean,
    ) {
        if (nonOwnedAccounts.isNotEmpty() && !isHouseholdMode) {
            throw ForbiddenException(
                "Cannot post to accounts not owned by current user in individual mode",
            )
        }
    }
}
